netjsonconfig/backends/openwrt/schema.py

```diff
@@ -386,10 +419,579 @@
+                                    "Must refer to one of the defined zone names. "
+                                    "For typical port forwards this usually is wan.",
+                                    "propertyOrder": 2,
+                                },
+                                "src_ip": {
+                                    "type": "string",
+                                    "title": "src_ip",
+                                    "description": "Match incoming traffic from the specified source ip "
+                                    "address.",
+                                    "format": "ipv4",
+                                    "propertyOrder": 3,
+                                },
+                                "src_dip": {
+                                    "type": "string",
+                                    "title": "src_dip",
+                                    "description": "For DNAT, match incoming traffic directed at the "
+                                    "given destination ip address. For SNAT rewrite the source address "
+                                    "to the given address.",
+                                    "format": "ipv4",
+                                    "propertyOrder": 4,
+                                },
+                                "src_mac": {
+                                    "type": "string",
+                                    "title": "src_mac",
+                                    "description": "Match incoming traffic from the specified MAC address.",
+                                    "pattern": mac_address_regex,
+                                    "minLength": 17,
+                                    "maxLength": 17,
+                                    "propertyOrder": 5,
+                                },
+                                "src_port": {
+                                    "type": "string",
+                                    "title": "src_port",
+                                    "description": "Match incoming traffic originating from the given source "
+                                    "port or port range on the client host.",
+                                    "pattern": port_range_regex,
+                                    "propertyOrder": 6,
+                                },
+                                "src_dport": {
+                                    "type": "string",
+                                    "title": "src_dport",
+                                    "description": "For DNAT, match incoming traffic directed at the given "
+                                    "destination port or port range on this host. For SNAT rewrite the "
+                                    "source ports to the given value.",
+                                    "pattern": port_range_regex,
+                                    "propertyOrder": 7,
```

Comment:
why don't we change this to an integer using minvalue and maxvalue to indicate the range?